paasta_tools/kubernetes_tools.py

```diff
@@ -2489,11 +2489,11 @@ def get_pod_template_spec(
             # character limit for k8s labels (63 chars)
             labels["paasta.yelp.com/deploy_group"] = self.get_deploy_group()
 
-        elif self.should_use_metrics_provider(METRICS_PROVIDER_PISCINA):
+        if self.should_use_metrics_provider(METRICS_PROVIDER_PISCINA):
             labels["paasta.yelp.com/deploy_group"] = self.get_deploy_group()
```

Comment:
We're now repeating this deploy_group label setting, but i guess no real harm :p

Reply:
!8ball we should just add this as a default label